### DISCUSSION TITLE
Number Input

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#235] [Component suggestions]
<img width="480" alt="visuele weergave van het number input component" src="https://github.com/nl-design-system/backlog/assets/545207/168ead44-8c15-49aa-b0c3-9b81fc41b406">

We willen graag meer leren over de Number Input component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1711382998610579)

---

## Naam
Number Input

## Doel
 Invoerveld voor een heel getal of een getal met decimalen.

## Ook bekend als
Numeric Input

## Gerelateerde componenten
[Checkbox Group](https://github.com/orgs/nl-design-system/discussions/223), [Date Input](https://github.com/orgs/nl-design-system/discussions/270), [File Input](https://github.com/orgs/nl-design-system/discussions/310), [Password Input](https://github.com/orgs/nl-design-system/discussions/285), [Radio Group](https://github.com/orgs/nl-design-system/discussions/222), [Range](https://github.com/orgs/nl-design-system/discussions/231), [Text Area](https://github.com/orgs/nl-design-system/discussions/297), [Text Input](https://github.com/orgs/nl-design-system/discussions/225), [Text Input Group](https://github.com/orgs/nl-design-system/discussions/296)

## Onderzoek
[Why the GOV.UK Design System team changed the input type for numbers - GOV.UK](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/)
Hanna Laakso legt uit waarom GOV.UK aanpassingen heeft gedaan aan de Number Input.

[Most web developers are using inaccessible number inputs on forms and here is how to fix it - Coywolf](https://www.coywolf.news/webdev/accessible-number-input/)
In dit artikel wordt besproken hoe veel webontwikkelaars onjuiste nummerinvoervelden gebruiken, en wordt aanbevolen om type=‘text’ met extra attributen te gebruiken om de toegankelijkheid voor iedereen te verbeteren.

[When to use the number input - Medium (Adam Silver)](https://medium.com/@adamsilverhq/when-to-use-the-number-input-b24e11ec550d)
In dit artikel wordt uitgelegd wanneer het gebruik van de ‘number’-invoer ongeschikt is, ondanks dat het veld vaak wordt gekozen voor gegevens zoals kaartnummers en vervaldatums, en wordt geadviseerd om ‘text’-invoer te gebruiken voor cijfers zonder rekenkundige waarde.

## Notities
[Number Input schets - NL Design System Figma Schetsboek](https://www.figma.com/file/fy08SZpZmqx6ljLwvA3Woe/NLDS---Schetsboek?type=design&node-id=1008-11796&mode=design&t=uoE8SgSwOmFkha5j-4)
Eerdere vragen en opmerkingen rondom dit component. Inclusief enkele schetsen.

**Comment by sergei-maertens** [comment]
## Open Formulieren

In Open Formulieren hebben we een aantal varianten:

- [Storybook - Form.io component](https://open-formulieren.github.io/open-forms-sdk/?path=/story/form-io-components-vanilla-number--number)
- [Storybook - Pure react](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/pure-react-components-forms-numberfield--docs)

---

### Screenshots / Varianten

**Storybook - Form.io**

Default
![Scherm­afbeelding 2024-03-27 om 11 22 58](https://github.com/nl-design-system/backlog/assets/545207/47654c40-502c-4660-aaa9-a302a5590396)

![Scherm­afbeelding 2024-03-27 om 11 23 20](https://github.com/nl-design-system/backlog/assets/545207/340297f0-d751-4719-909f-96bb773b62a9)

With suffix
![Scherm­afbeelding 2024-03-27 om 11 23 06](https://github.com/nl-design-system/backlog/assets/545207/fe3518e2-5c80-4496-9e8c-ab56ea16f506)

![Scherm­afbeelding 2024-03-27 om 11 23 35](https://github.com/nl-design-system/backlog/assets/545207/e2dbbc49-3bcf-4cb5-9ddf-ca2466198988)

**Storybook - Pure react**

Small number inputs
![Scherm­afbeelding 2024-03-27 om 11 29 08](https://github.com/nl-design-system/backlog/assets/545207/bdffde18-749a-4bc1-bea0-0fed5bbd2950)

![Scherm­afbeelding 2024-03-27 om 11 29 59](https://github.com/nl-design-system/backlog/assets/545207/ebd72f33-4e25-43c4-8688-a26446688a2e)

Free form number fields
![Scherm­afbeelding 2024-03-27 om 11 29 16](https://github.com/nl-design-system/backlog/assets/545207/5da9a9a4-c640-47d3-b96d-6c82828399dc)

![Scherm­afbeelding 2024-03-27 om 11 30 17](https://github.com/nl-design-system/backlog/assets/545207/8ee4d541-38ce-4d5e-b3bd-e56d2bfebc1e)

---

### Notities
- Merk op dat het decimale scheidingsteken meeverandert met de actieve locale (dus <kbd>,</kbd> in NL en <kbd>.</kbd> in EN). Hetzelfde gebeurt bij het groeperen van de duizendtallen.
- Uitgebreide documentatie beschikbaar.

**Comment by jeffreylauwers** [comment]
## Amsterdam

### Notities

Geen apart component gevonden. Number input lijkt [Text input](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8743963) stijl te volgen.

**Comment by jeffreylauwers** [comment]
## Den Haag

[Storybook](https://nl-design-system.github.io/denhaag/?path=/story/open-forms-number--default) (op basis van Open Forms)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-03-27 om 11 09 06](https://github.com/nl-design-system/backlog/assets/545207/5909409c-c558-476d-87a1-0e828f0f59c3)

With prefix
![Scherm­afbeelding 2024-03-27 om 11 09 13](https://github.com/nl-design-system/backlog/assets/545207/74e69f97-a727-4522-a462-fd035d70e45b)

With suffix
![Scherm­afbeelding 2024-03-27 om 11 09 22](https://github.com/nl-design-system/backlog/assets/545207/59c27c20-2fb3-4978-8105-b80a08797725)

---

### Notities
- Number input lijkt [Text input](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8744057) stijl te volgen.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-input-number--input-number&viewMode=story)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-03-27 om 11 12 12](https://github.com/nl-design-system/backlog/assets/545207/dd8f7172-2016-4b5e-8136-26a0d0eb04f1)

---

### Notities
- Geen mogelijkheid om vrij in te vullen.
- Eerder 'Stepper' component.

**Comment by jeffreylauwers** [comment]
## DUO

### Notities

Number input functionaliteit lijkt onderdeel te zijn van het [Text input](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8744362) component.

> **Wanneer gebruiken**
> Gebruik een text input in een formulier als gebruikers een antwoord moet invoeren dat op een enkele regel moet passen. Bijvoorbeeld: een e-mailadres, **een getal**, een telefoonnummer.

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

[Patternlab - Forms and Search](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search/index.html)

---

### Screenshots / Varianten

**Patternlab**

Default
![Scherm­afbeelding 2024-03-27 om 11 41 08](https://github.com/nl-design-system/backlog/assets/545207/681f754b-a4e2-4504-8f0d-ad36cf1ac2d2)

![Scherm­afbeelding 2024-03-27 om 11 41 16](https://github.com/nl-design-system/backlog/assets/545207/1d67e5b9-07a1-4575-8533-2608c8af08a8)

---

### Notities
- Number input lijkt [Text input](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8746808) stijl te volgen.

**Comment by jeffreylauwers** [comment]
## Rotterdam

### Notities

Geen apart component gevonden. Number input lijkt [Text input](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8747081) stijl te volgen.

**Comment by jeffreylauwers** [comment]
## RVO

### Notities

Geen apart component gevonden. Number input lijkt [Text input](https://github.com/orgs/nl-design-system/discussions/225#discussioncomment-8747275) stijl te volgen.

**Comment by rianrietveld** [comment]
Graag wil ik hier mijn reserve uitpreken voor het gebruik van `type="number"`.
De NLDS richtlijn bij formulieren > [Zorg dat iedereen een formulierelement kan bedienen of geef een alternatief](https://nldesignsystem.nl/richtlijnen/formulieren/wanneer-welk-form-element/iedereen-kan-invullen) > Input type=“number” versus inputmode=“numeric” geeft een voorkeur voor `inputmode="numeric" ` aan.
GOV.UK heeft hier onderzoek naar gedaan en vindt [type=number qua gebruikersvriendelijkheid te foutgevoelig](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/).

En over een bedrag met decimalen:
Het gebruik van een komma/punt kan afhankelijk zijn van de taal van de browser en kan tot grote fouten leiden. Zo kun je per ongeluk ineens hele andere bedragen invullen (of onbegrijpelijke foutmeldingen krijgen) als je browser van een komma en punt maakt. Of als een Amerikaan een bedrag invult zoals deze gewend.

Ik ben het ermee eens dat dit een component moet worden, maar dan ook graag bovenstaande als overweging meenemen.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Number Input op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=number+input)
